### PR TITLE
Add removal of flat

### LINKED ISSUE #1533: Use of flat breaks vscode debugging
## Describe the bug

- vscode debugging doesn't like `flat`.

### To reproduce

1. Try to debug in vscode

### Expected behaviour

Should be able to debug in vscode

### Version and device info

- App version: 3.0+
- Tablet model: All
- OS version: All

### Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1533 

## Change summary

- Removes flat call

## Testing

- [ ] Running vscode debug is possible
- [ ] Realm explorer page works (note it had a couple warnings that are unrelated)

### Related areas to think about

N/A

### CODE REVIEW

**Comment by Chris-Petty** [comment]
Tested, works 👍 Also I think the code is clearer without map.

**Comment by Chris-Petty** [comment]
This started happening though 
image

**Comment by joshxg** [comment]
hmm didn't get that and can't really see a way that you could! maybe ravi or sworup can give it a spin

**Comment by raviSussol** [comment]
@joshxg Im also getting that error while running debugger in the vscode.

**Comment by joshxg** [comment]
@raviSussol @Chris-Petty anymore details/help please? I can't recreate this

File: src/pages/RealmExplorer.js [hunk 1]
```diff
@@ -43,24 +43,25 @@ const TYPE_MAPPINGS = {
...
 
-const getRealmObjects = ({ schema: objectSchemas }) =>
-  objectSchemas.map(({ schema: objectSchema }) => objectSchema.name);
+const getRealmObjects = ({ schema: objectSchemas }) => objectSchemas.map(({ name }) => name);
 
 const getRealmObjectsFields = ({ schema: objectSchemas }) =>
   objectSchemas
-    .map(({ schema: objectSchema }) => {
+    .map(objectSchema => {
       const { name, properties } = objectSchema;
```
Comment (raviSussol) [inline src/pages/RealmExplorer.js]: `schema` keyword needs to be attached to schema value while loading Realm schema. That resolves an error and runs ok in debugger mode in vscode.

`.map(({ schema: objectSchema }) => {`

**Comment by joshxg** [comment]
@Chris-Petty @raviSussol 

I ended up being able to recreate the error you're showing when the debugger was not running. However, if I used the code above that ravi suggests, it would throw an error in the debugger.

I've added f18d5ac - I don't know what's happening and this is my hack to fix it - if anyone wants to have a go feel free.. possibly you could try running the code in the component function, could be some weird synchronous importing module behaviour

**Comment by Chris-Petty** [comment]
Yea I should have clarified, it was specifically when I turned the debugger off this error started.

**Comment by raviSussol** [comment]
@joshxg 
> I ended up being able to recreate the error you're showing when the debugger was not running. However, if I used the code above that ravi suggests, it would throw an error in the debugger.

Can you attach the screenshot of error(s) that you found? It works fine on me both in (without debug `react-native run-android` and with debugging by vscode)

With https://github.com/openmsupply/mobile/commit/f18d5acc9741048a2223f4f821c26c5b1712eb11 Im getting errors when running `react-native run-android`

**Comment by joshxg** [comment]
This back and forward is getting frustrating. If you're going to comment on the PR that you are getting errors, please be more specific, otherwise I cannot make any changes. Saying "I am getting errors" is not helpful, as it can be safe for you to assume that I am not, so how can I do anything.

**Comment by joshxg** [comment]
With your change I get the error here in debug mode
image


Where as without debugging, it runs fine.

The change in the commit is literally to use the field which your recommendation suggests, but if it is undefined, use a different variable. If an error was thrown from that change, it would have been thrown with the change you suggested?

The problem seems to be that in debug mode, the value passed does not have a `schema` property, where as in non-debug, it does. It's strange behaviour that I don't really have time to look more into unfortunately

**Comment by raviSussol** [comment]
@joshxg I think Im familiar with the error above. Can you try with below steps:
- Clean the gradle - `cd android && ./gradlew clean`
- Clear the yarn cache - `yarn cache clean --force`
- Uninstall the installed mSupply mobile app before you run debugger in the vscode.

If still those don't work then try killing current or all node instance(s) by running `killall -9 node` for current or `killall node` for all.

**Comment by joshxg** [comment]
@raviSussol hasn't worked.

I have just made your change b541c31 as I don't use this anyway 🤷‍♂

**Pull request merged**